bin/fuel-core/src/cli/run/graphql.rs

```diff
@@ -28,6 +28,14 @@ pub struct GraphQLArgs {
     #[clap(long = "graphql-max-directives", default_value = "10", env)]
     pub max_queries_directives: usize,
 
+    /// The max number of directives in the query.
+    #[clap(
+        long = "graphql-max-concurrent-queries",
+        default_value = "usize::MAX",
```

Comment:
Maybe the default should be something reasonable like 1024?

Reply:
Sure, I'm curious why 1024 is a reasonable number here though? Is it your gut feeling or do we have any insights in the expected load on nodes that we can base this number on?

My reasoning about setting `usize::MAX` (though I have corrected it to `Semaphore::MAX_PERMITS` now) is that it is less intrusive for the node operators compared to having an arbitrary limit imposed on them which they may not be aware of.

I can also imagine that some node operators may want to configure this type of limit directly on their load balancer, in which case it's nice not to have a default limit set on our end.

That being said, you have much more insight than I in how this is going to be run once released so I trust your judgement. What are your thoughts?

Reply:
we should probably base it on a worst case estimate of query load against our recommended node specs in the docs:

https://docs.fuel.network/guides/running-a-node/#hardware-requirements

@xgreenx should be able to help with coming up with the worst case ball-park figure

Reply:
Alright, I've set it to 1024 for now and added two integration tests. Let me see if I can find some numbers anywhere, otherwise I agree that if we want to be conservative 1024 feels like a good number. But I guess we're looking for something like $\frac{8\text{GB}}{m}$ where $m$ is the maximum amount of memory we expect a single query to require.

Reply:
So I've set the default as 1024, and also marked this change as breaking now.